dask/blockwise.py

```diff
@@ -423,8 +423,8 @@ def __init__(
             name = dep
             if isinstance(dep, BlockwiseDep):
                 name = tokenize(dep)
-                self.io_deps[name] = dep
-                self.numblocks[name] = dep.numblocks
+                self.io_deps[name] = dep  # type: ignore
+                self.numblocks[name] = dep.numblocks  # type: ignore
```

Comment:
This can be fixed by annotating above as a `MutableMapping`. But that also should be telling us that we should be making a copy of `numblocks` as a dict internally.